File: examples/simp/CPP/predictor/predictor3.cpp
#include <linkpred.hpp>
#include <iostream>

using namespace LinkPred::Simp;

int main() {
#ifdef LINKPRED_WITH_MLPACK
	int k = 10;
	Predictor p;
	p.loadnet("Zakarays_Karate_Club.edges");
	{
		auto esv = p.predTopECL(k, "DPW", "FFN");
		std::cout << "Top " << k << " using DPW-FFN\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "DPW", "LSVM");
		std::cout << "Top " << k << " using DPW-LSVM\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "DPW", "LGR");
		std::cout << "Top " << k << " using DPW-LGR\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "DPW", "NVB");
		std::cout << "Top " << k << " using DPW-NVB\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}

	{
		auto esv = p.predTopECL(k, "HMSM", "FFN");
		std::cout << "Top " << k << " using HMSM-FFN\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "HMSM", "LSVM");
		std::cout << "Top " << k << " using HMSM-LSVM\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "HMSM", "LGR");
		std::cout << "Top " << k << " using HMSM-LGR\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "HMSM", "NVB");
		std::cout << "Top " << k << " using HMSM-NVB\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}

	{
		auto esv = p.predTopECL(k, "LVS", "FFN");
		std::cout << "Top " << k << " using LVS-FFN\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LVS", "LSVM");
		std::cout << "Top " << k << " using LVS-LSVM\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LVS", "LGR");
		std::cout << "Top " << k << " using LVS-LGR\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LVS", "NVB");
		std::cout << "Top " << k << " using LVS-NVB\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}

	{
		auto esv = p.predTopECL(k, "LIN", "FFN");
		std::cout << "Top " << k << " using LIN-FFN\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LIN", "LSVM");
		std::cout << "Top " << k << " using LIN-LSVM\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LIN", "LGR");
		std::cout << "Top " << k << " using LIN-LGR\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "LIN", "NVB");
		std::cout << "Top " << k << " using LIN-NVB\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}

	{
		auto esv = p.predTopECL(k, "N2V", "FFN");
		std::cout << "Top " << k << " using N2V-FFN\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "N2V", "LSVM");
		std::cout << "Top " << k << " using N2V-LSVM\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "N2V", "LGR");
		std::cout << "Top " << k << " using N2V-LGR\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
	{
		auto esv = p.predTopECL(k, "N2V", "NVB");
		std::cout << "Top " << k << " using N2V-NVB\n";
		for (auto it = esv.begin(); it != esv.end(); ++it) {
			std::cout << it->i << "\t" << it->j << "\t" << it->score
					<< std::endl;
		}
	}
#endif

	return 0;
}

File: examples/simp/Python/predictor/predictor3.py
import LinkPredPython as lpp
k = 10;
p = lpp.Predictor();
p.loadnet("Zakarays_Karate_Club.edges");

esv = p.predTopENC(k, "DPW", "FFN");
print("Top " + str(k) + " using DPW-FFN");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "DPW", "LSVM");
print("Top " + str(k) + " using DPW-LSVM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "DPW", "LGR");
print("Top " + str(k) + " using DPW-LGR");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "DPW", "NVB");
print("Top " + str(k) + " using DPW-NVB");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));


esv = p.predTopENC(k, "HMSM", "FFN");
print("Top " + str(k) + " using HMSM-FFN");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "HMSM", "LSVM");
print("Top " + str(k) + " using HMSM-LSVM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "HMSM", "LGR");
print("Top " + str(k) + " using HMSM-LGR");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "HMSM", "NVB");
print("Top " + str(k) + " using HMSM-NVB");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));


esv = p.predTopENC(k, "LVS", "FFN");
print("Top " + str(k) + " using LVS-FFN");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LVS", "LSVM");
print("Top " + str(k) + " using LVS-LSVM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LVS", "LGR");
print("Top " + str(k) + " using LVS-LGR");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LVS", "NVB");
print("Top " + str(k) + " using LVS-NVB");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));


esv = p.predTopENC(k, "LIN", "FFN");
print("Top " + str(k) + " using LIN-FFN");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LIN", "LSVM");
print("Top " + str(k) + " using LIN-LSVM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LIN", "LGR");
print("Top " + str(k) + " using LIN-LGR");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "LIN", "NVB");
print("Top " + str(k) + " using LIN-NVB");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));


esv = p.predTopENC(k, "N2V", "FFN");
print("Top " + str(k) + " using N2V-FFN");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "N2V", "LSVM");
print("Top " + str(k) + " using N2V-LSVM");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "N2V", "LGR");
print("Top " + str(k) + " using N2V-LGR");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

esv = p.predTopENC(k, "N2V", "NVB");
print("Top " + str(k) + " using N2V-NVB");
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

